.github/workflows/main.yml

```diff
@@ -24,15 +24,10 @@ jobs:
         suite:
           - 'agent'
           - 'agent-only'
+          - 'gpu-agent'
         chef_version:
           - '16'
           - '17'
-        nvidia_support:
-          - true
-          - false
-        exclude:
-          - suite: 'agent-only'
-            nvidia_support: true
```

Comment:
This refactoring to use a dedicated test kitchen suite for the gpu agent also decomplicates our test matrix expression. 
Here's an enumeration of the old matrix combinations with their new counterparts (excludes marked with ~~strikeout~~):
{suite: agent, chef_version: 16, nvidia_support: true} -> {suite: gpu-agent, chef_version: 16}
{suite: agent, chef_version: 16, nvidia_support: false} -> {suite: agent, chef_version: 16}
{suite: agent, chef_version: 17, nvidia_support: true} -> {suite: gpu-agent, chef_version: 17}
{suite: agent, chef_version: 17, nvidia_support: false} -> {suite: agent, chef_version: 17}
~~{suite: agent-only, chef_version: 16, nvidia_support: true}~~
{suite: agent-only, chef_version: 16, nvidia_support: false} -> {suite: agent-only, chef_version: 16}
~~{suite: agent-only, chef_version: 17, nvidia_support: true}~~
{suite: agent-only, chef_version: 17, nvidia_support: false} -> {suite: agent-only, chef_version: 17}

which is 6 total jobs. After this change there are the following combinations: